### PR TITLE
Install gateway-proxy in separate namespace from Gloo Edge Control Plane

### PR DESCRIPTION
**Pull request by mkazin**
# Description

Allows setting the namespace of amy named proxy (gatewayProxies.NAME) using a new 

Draft notes:
* This change does not create any of the specified non-control-plane namespaces. These must  be created manually.

# Context

This change allows Gloo Edge to be deployed in a more customized configuration. This is required when integrating proxies into Gloo Mesh Workspaces which use namespaces for isolation.
 
# Testing
Testing was done manually, verifying that XDS configuration reached proxies in both gloo-system and a custom  namespace.

Configuration used in testing (based on the [multi-gw deployment guide](https://github.com/solo-io/gloo/blob/master/docs/content/installation/advanced_configuration/multi-gw-deployment.md))
```
  gatewayProxies:
    publicGw: # Proxy name for public access (Internet facing)
      disabled: false # overwrite the "default" value in the merge step
      kind:
        deployment:
          replicas: 2
      service:
        kubeResourceOverride: # workaround for https://github.com/solo-io/gloo/issues/5297
          spec:
            ports:
              - port: 443
                protocol: TCP
                name: https
                targetPort: 8443
            type: LoadBalancer
      gatewaySettings:
        customHttpsGateway: # using the default HTTPS Gateway
          virtualServiceSelector:
            gateway-type: public # label set on the VirtualService
        disableHttpGateway: true # disable the default HTTP Gateway
    corpGw: # Proxy name for private access (intranet facing)
      disabled: false # overwrite the "default" value in the merge step
      namespace: corp-ns
      service:
        httpPort: 80
        httpsFirst: false
        httpsPort: 443
        httpNodePort: 32080 # random port to be fixed in your private network
        type: NodePort
      # Creates corp-gw-config-dump-service
      readConfig: true
      readConfigMulticluster: true
      gatewaySettings:
        customHttpGateway: {} # using the default HTTP Gateway
        disableHttpsGateway: true # disable the default HTTPS Gateway
    gatewayProxy:
      disabled: false # disable the default gateway-proxy deployment and its 2 default Gateway CRs
```

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/6295

### CODE REVIEW

File: install/test/fixtures.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ node:
   id: "{{.PodName}}.{{.PodNamespace}}"
   metadata:
     # role's value is the key for the in-memory xds cache (projects/gloo/pkg/xds/envoy.go)
-    role: "{{.PodNamespace}}~gateway-proxy"
+    role: "gloo-system~gateway-proxy"
```
Comment (inFocus7) [inline install/test/fixtures.go:20]: Probably a dumb q, but why are we hardcoding `.PodNamespace` to `gloo-system` now? (extends to this and all other areas where `.PodNamespace` was replaced)

And should we update the `id` a few lines above to match this change as well?

Comment (nfuden) [inline install/test/fixtures.go:20]: is this change necessary?

**Review by nfuden — COMMENTED** [review]
We should have a test to prove that this outputs correctly for 2 proxies in different namespaces

**Comment by inFocus7** [comment]
Also agree with Nathan on testing that different proxies w/ different namespaces works! 💯

File: install/test/fixtures.go (hunk 1 above)
Comment (mkazin) [reply install/test/fixtures.go:20]: Those are good questions!

Q1:
1. This is a test fixture, so it's not so much hard-coding a value, but specifying the expected value.
2. .PodNamespace is a special value which is not translated by Helm, but rather by Envoy => we're looking at the expected Envoy config here.
3. This *was* the expected value when the namespace of gateway-proxies was identical to that of the pod they ran in, which also matched the namespace of the gloo pod.
4. That's no longer the case, as this PR allows for the deployment of a gateway-proxy in other namespaces.
5. However, we don't want to keep using the pod's namespace, as this represents the key of the xds cache under which the proxy's config is stored
6. To avoid confusion and possible upgrade issues we decided not to change the xds keys, and leaving them with whatever was used for the writeNamespace (gloo's namespace). In these tests, that's "gloo-system".

Q2: no, I believe the id is correct, as it identifies the envoy node in its installed namespace.

**Review by Nadine2016 — APPROVED** [review]

File: install/test/fixtures.go (hunk 1 above)
Comment (mkazin) [reply install/test/fixtures.go:20]: I think so. Please see [this response](https://github.com/solo-io/gloo/pull/6505#discussion_r883906230) I left to Fabian's similar question.

**Review by nfuden — APPROVED** [review]
